irods_browser_app/views.py

```diff
@@ -0,0 +1,137 @@
+import json
+import os
+from django.http import HttpResponse, HttpResponseRedirect
+from irods.session import iRODSSession
+from irods.exception import CollectionDoesNotExist
+from hs_core import hydroshare
+
+irods_sess = None
+
+def search_ds(coll):
+    store = {}
+    file = []
+    folder = []
+    if coll.data_objects:
+        for files in coll.data_objects:
+            file.append(files.name)
+    if coll.subcollections:
+        for folders in coll.subcollections:
+            folder.append(folders.name)
+
+    store['files'] = file
+    store['folder'] = folder
+    return store
+
+# Create your views here.
+def login(request):
+    if request.method == 'POST':
+        port = int(request.POST['port'])
+        user = str(request.POST['user'])
+        password = str(request.POST['password'])
+        zone = str(request.POST['zone'])
+        host = str(request.POST['host'])
+        datastore = "/%s/home/%s" % (zone, user)
+
+        response_data = {}
+
+        global irods_sess
+        irods_sess = iRODSSession(user=user, password=password, zone=zone, host=host, port=port)
+
+        try:
+            irods_sess.collections.get(datastore)
+        except CollectionDoesNotExist:
+            request.session['irods_loggedin'] = False
+            request.session['login_message'] = 'iRODS login failed'
+            request.session['irods_file_name'] = ''
+            response_data['irods_loggedin'] = False
+            response_data['login_message'] = 'iRODS login failed'
+            response_data['irods_file_name'] = ''
+            response_data['error'] = "iRODS collection does not exist"
+            return HttpResponse(
+                json.dumps(response_data),
+                content_type="application/json"
+            )
+        else:
+            request.session["user"] = user
+            request.session["datastore"] = datastore
+            request.session["password"] = password
```

Comment:
It isn't just that globals aren't a good idea and you should find an alternative: this implementation is already broken. It will not work in production nor will it be safe.

My expectation was that these views are AJAX views, and you both login and trigger the iRODS operations from AJAX requests initiated from the resource creation and edit pages. Is that right?